frontend/src/components/InlineStatusEdit.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <v-menu offset-y :style="{ height: '100%', width: '100%', display: 'inherit' }">
+    <p class="ma-0 text-no-wrap" slot="activator" :style="{ height: '25px', width: '40px' }">
+      {{ status }}  
+    </p>
+    <div>
+      <v-list>
+        <v-list-tile @click="handleChange(item)" v-for="(item, index) in statusItems" :key="index">
+          <v-list-tile-title>{{ item }}</v-list-tile-title>
+        </v-list-tile>
+      </v-list>
+    </div>
+  </v-menu>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+import { findObjectValue } from "@/utility";
+
+@Component({
+  name: "InlineStatusEdit"
+})
+export default class InlineStatusEdit extends Vue {
+  @Prop() item!: any;
+  @Prop() statusItems!: string[];
+  @Prop() keyPath!: string;
+
+  get status() {
+    const { item } = this;
+    return findObjectValue(item, "status");
+  }
+
+  handleChange(e: string) {
+    const { item } = this;
+    
+    //Object that is use in making of firebase path URL to save data in database.
+    let path: any = {};
+    path["keyPathId"] = item[".key"] ? item[".key"] : "";
+    path["keyPath"] = this.keyPath ? this.keyPath : "";
+    path["itemPath"] = "status";
+    this.$emit("save", item, path, e, { newValue: e, itemPath: "status" });
```

Comment:
ok

Reply:
`itemPath` still uses the hard coded `"status"`. Is this working as intended or does this need to be replaced with `this.value`?

Reply:
It is working and it's use hardcoded status because this component is for status and I have set logic according to this...so don't worry about this.

Reply:
Some paths to the status are nested like `restoration.status`. Does this work? Based on the save method it seems it would need the full path.

Reply:
Yes it work.

Reply:
It work I have coded for this

Reply:
I believe this works currently because of the way firebase works. When it receives an update it also updates the client. If the connection is slow the user will not see the new data instantly. Currently for instant updates the exact path to the updated item must be provided. You can see what I mean if you take away the firebase part of the save and just leave the state update. You can also see in this example https://repl.it/@dadimscared/TraumaticShockedObjects